### ISSUE TITLE
Dashboard rook module and custom namespace not working

### ISSUE THREAD

**Issue by pasztorl** [rook/rook#13498]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

If I configure the rook and rook-cluster charts to use an alternate namespace the mgr pod not see it.
Inside the "mgr" container the environment settings looks good, because the ROOK_OPERATOR_NAMESPACE points to the  right namespace, but the mgr still use the default "rook-ceph". In the test system this namespace not exist so it got 404.

```
debug 2024-01-03T20:47:55.584+0000 7f74b3039700  0 [rook ERROR orchestrator._interface] (404)
Reason: Not Found
HTTP response headers: HTTPHeaderDict({'Audit-Id': 'e7e3cccc-483d-42f3-9bd3-cde885334e40', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'X-Kubernetes-Pf-Flowschema-Uid': '7378126a-e88c-42cc-9b56-eb30463ede30', 'X-Kubernetes-Pf-Prioritylevel-Uid': '6be186d5-75b7-4618-a689-aac90f8efe3c', 'Date': 'Wed, 03 Jan 2024 20:47:55 GMT', 'Content-Length': '230'})
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"configmaps \"rook-ceph-operator-config\" not found","reason":"NotFound","details":{"name":"rook-ceph-operator-config","kind":"configmaps"},"code":404}

Traceback (most recent call last):
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 125, in wrapper
    return OrchResult(f(*args, **kwargs))
  File "/usr/share/ceph/mgr/rook/module.py", line 229, in get_inventory
    discovered_devs = self.rook_cluster.get_discovered_devices(host_list)
  File "/usr/share/ceph/mgr/rook/rook_cluster.py", line 754, in get_discovered_devices
    op_settings = self.coreV1_api.read_namespaced_config_map(name="rook-ceph-operator-config", namespace='rook-ceph').data
  File "/usr/lib/python3.6/site-packages/kubernetes/client/api/core_v1_api.py", line 18390, in read_namespaced_config_map
    (data) = self.read_namespaced_config_map_with_http_info(name, namespace, **kwargs)  # noqa: E501
  File "/usr/lib/python3.6/site-packages/kubernetes/client/api/core_v1_api.py", line 18481, in read_namespaced_config_map_with_http_info
    collection_formats=collection_formats)
  File "/usr/lib/python3.6/site-packages/kubernetes/client/api_client.py", line 345, in call_api
    _preload_content, _request_timeout)
  File "/usr/lib/python3.6/site-packages/kubernetes/client/api_client.py", line 176, in __call_api
    _request_timeout=_request_timeout)
  File "/usr/lib/python3.6/site-packages/kubernetes/client/api_client.py", line 366, in request
    headers=headers)
  File "/usr/lib/python3.6/site-packages/kubernetes/client/rest.py", line 241, in GET
    query_params=query_params)
  File "/usr/lib/python3.6/site-packages/kubernetes/client/rest.py", line 231, in request
    raise ApiException(http_resp=r)
kubernetes.client.rest.ApiException: (404)
```

**Expected behavior:**

The dashboard's rook module use the ROOK_OPERATOR_NAMESPACE environment variable.

**How to reproduce it (minimal and precise):**

In rook-cluster helm chart values:

```
operatorNamespace: xxx
mgr:
    count: 1
    allowMultiplePerNode: false
    modules:
      - name: rook
        enabled: true
      - name: pg_autoscaler
        enabled: true

```

ROOK chart versions:
rook-ceph-v1.13.1
rook-ceph-cluster-v1.13.1

**Comment by travisn** [comment]
This should be fixed in the latest ceph v18.2.1. Please update to that ceph version. The version installed by default in the helm chart was still v17.2.6, which will be fixed in the next Rook patch release with #13499.

**Comment by pasztorl** [comment]
18.2.1 works, thanks!
